wire-ios/Wire-iOS/Sources/UserInterface/E2EIdentity/E2EINotificationActionsHandler.swift

```diff
@@ -174,39 +179,45 @@ final class E2EINotificationActionsHandler: E2EINotificationActions {
             self.isUpdateMode = false
         }
 
-        targetVC.present(successScreen, animated: true)
+        presentScreen(viewController: successScreen)
     }
 
-    private let durationFormatter: DateComponentsFormatter = {
-        let formatter = DateComponentsFormatter()
-        formatter.allowedUnits = [.day, .hour, .minute]
-        formatter.unitsStyle = .abbreviated
-        return formatter
-    }()
+    @MainActor
+    private func presentScreen(viewController: UIViewController) {
+        let vc = UIApplication.shared.topmostViewController(onlyFullScreen: false) ?? targetVC
+        vc.present(viewController, animated: true)
```

Comment:
question: is this the wanted behaviour that we present just on top of everything?

My high level idea would be rather to replace the current window with a simple new screen, where the user has no possibilities for any interactions, so to say a different area of the app. Once everything is validated again, we would move him back to where he left.

Reply:
it's better than to fail presenting the view.

We have something similar to your idea that we could use `BlockerViewController` with the screencurtain (UIWindow) though it would differ from what design provided.

I'll keep as it is for now